src/vdaf.rs

```diff
@@ -123,7 +125,7 @@ pub trait Vdaf: Clone + Debug {
     type InputShare: Clone + Debug + ParameterizedDecode<Self::VerifyParam> + Encode;
 
     /// An output share recovered from an input share by an Aggregator.
-    type OutputShare: Clone + Debug;
+    type OutputShare: Clone + Debug + Decode + Encode;
```

Comment:
I think putting `Encode/Decode` on `struct OutputShare` is sound. However I'm not sure it makes sense to put this bound into `trait Vdaf`. Generally what's in the trait corresponds to requirements of the VDAF specification, and there's never a need for one VDAF implementation to transmit an output share to another, so draft-patton-cfrg-vdaf doesn't specify how to encode those. Then again, VDAF doesn't say anything about `Clone + Debug`, either, so maybe we don't have to treat this trait definition as equivalent to the VDAF spec. I'm curious to hear @cjpatton's opinion.

I'll note that an alternative to putting this bound on `trait Vdaf` is to declare it in your code that is generic over `Vdaf` (or over `Aggregator`). For instance:

```
fn do_thing<V>() 
where
    V: Vdaf,
    Vdaf::OutputShare: Decode + Encode,
{ ... }
```

The above trait bound holds for `struct Prio3` and `struct Poplar1` because they both have `OutputShare = vdaf::OutputShare`, which does implement `Encode/Decode`, but you wouldn't be able to use any implementation of `Vdaf` whose `OutputShare` associated type doesn't implement those traits.

Reply:
Agreed, part of this PR's review is to determine if it's acceptable to add `Encode`/`Decode` here or not:

> I also added a requirement that Vdaf::OutputShare implement Encode & Decode, following the lead of a few other types related to Vdaf. But I could drop this at the cost of slightly complexifying the trait bounds in implementations--I'm not opinionated here.

We have to declare many trait bounds over what is guaranteed by `Vdaf` in our implementation (e.g. from WIP: https://github.com/divviup/janus/compare/aggregate#diff-89c14b8f557baa95520c0bf22f9aeaa0c885731a65acee9f629ce7937208663dR108-R113), so determining the correct "line" for what goes in `libprio-rs` vs what goes in implementations using the library would be helpful.

Reply:
Removed Decode/Encode bound on OutputShare; will move to the implementation instead.